File: src/main/java/org/artembogomolova/demo/webapp/controller/PublicController.java
package org.artembogomolova.demo.webapp.controller;

import java.util.Arrays;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class PublicController {

  private static final String ROOT_URL = "/";
  private static final String INDEX_PAGE_URL = "/index";
  public static final String LOGIN_URL = "/login";
  public static final String TEMPLATE_RESOURCES_URL = "/templates/**";
  public static final String CSS_RESOURCES_URL = "/css/**";
  public static final String WEBJARS_RESOURCES_URL = "/webjars/**";
  private static final String PUBLIC_PAGE_TEMPLATE_PREFIX = "fragments/public";
  private static final String[] UNSECURED_RESOURCES = {
      /*public resources*/
      ROOT_URL, INDEX_PAGE_URL,
      /*resources*/
      TEMPLATE_RESOURCES_URL, CSS_RESOURCES_URL, WEBJARS_RESOURCES_URL};

  public static String[] getUnsecuredResources() {
    return Arrays.copyOf(UNSECURED_RESOURCES, UNSECURED_RESOURCES.length);
  }

  @GetMapping({ROOT_URL, INDEX_PAGE_URL})
  public String getIndexPage() {
    return PUBLIC_PAGE_TEMPLATE_PREFIX + INDEX_PAGE_URL;
  }

  @GetMapping(LOGIN_URL)
  public String getLoginUrl() {
    return PUBLIC_PAGE_TEMPLATE_PREFIX + LOGIN_URL;
  }
}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/controller/PublicController.kt
package org.artembogomolova.demo.webapp.main.controller

import org.springframework.http.MediaType
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ResponseBody

@Controller
class PublicController {
    @GetMapping(*[ROOT_URL, INDEX_PAGE_URL], produces = [MediaType.TEXT_PLAIN_VALUE])
    @ResponseBody
    fun indexPage() = "Hello!"


    companion object {
        private const val ROOT_URL = "/"
        private const val INDEX_PAGE_URL = "/index"
        private val UNSECURED_RESOURCES = arrayOf(
            /*public resources*/
            ROOT_URL, INDEX_PAGE_URL,  /*resources*/
        )
        val unsecuredResources: Array<String?>
            get() = UNSECURED_RESOURCES.copyOf(UNSECURED_RESOURCES.size)
    }
}